docs/rfcs/20220209_limit_connections_rate.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: Ability to limit connection creation rate per shard
+- Status: in-progress
+- Start Date: 2022-02-09
+- Authors: Vadim Plakhtinskiy
+- Issue: 
+
+# Executive Summary
+
+- Set a limit on the rate with which the shard accepts new connections
+- Set a limit on connection creation rate per IP
+
+
+Similar to KIP-612, adapted to Redpanda's thread per core model
+
+## What is being proposed
+
+New configuration properties (For start we choose global config for connection rate):
+- kafka_max_connection_creation_rate (optional integer, default unset)
+- kafka_max_connection_creation_rate_ovverides (string, default "")
+
+
+When a connection is accepted which exceeds one of the configured
+bounds, it is blocking before reading or writing anything to the socket.
+
+
+## Motivation
+
+* Avoid hitting system resource limits (for example CPU)
+* Avoid geting connections storms
+* Allow broker doing useful work instead of handling high connection creation rate
+
+## How
+
+If both configuration properties are unset, then no action is taken: this
+preserves the existing unlimited behavior.
+ 
+We will track connection rate on each shard independently (without cross-core communication).
+ 
+We will store semaphore to count connection in current second. For overrides we will store hash table
```

Comment:
I would suggest using a rate limiting algorithm here. f.e. token bucket based rate limiter.

Reply:
Would be great to do mention that